errai-bus/src/main/java/org/jboss/errai/bus/server/service/ErraiServiceSingleton.java

```diff
@@ -30,29 +30,33 @@ private ErraiServiceSingleton() {
   private static Set<ErraiInitCallback> callbackList = Collections.synchronizedSet(new HashSet<ErraiInitCallback>());
 
   private static final Object monitor = new Object();
-  private static volatile ErraiService service = new ErraiServiceProxy();
+  private static volatile ErraiServiceProxy initProxy = new ErraiServiceProxy();
+  private static volatile ErraiService service;
 
   public static ErraiService initSingleton(final ErraiServiceConfigurator configurator) {
     synchronized (monitor) {
       if (isInitialized()) throw new IllegalStateException("service already set into singleton");
-      ErraiServiceProxy proxy = (ErraiServiceProxy) service;
+
       service = ErraiServiceFactory.create(configurator);
-      proxy.closeProxy(service);
+      if(!initProxy.isClosed()) {
```

Comment:
Yeah you are right actually, I wasn't sure if the service would retain a reference since I'm not sure how Guice loads the dependencies, so I assumed the proxy wasn't really required after the initial load.